aineko/extras/connectors/rest.py

```diff
@@ -0,0 +1,152 @@
+# Copyright 2023 Aineko Authors
+# SPDX-License-Identifier: Apache-2.0
+"""Extra module for connecting to a REST endpoint."""
+
+import json
+import time
+from typing import Any, Dict, Optional
+
+import requests
+from pydantic import BaseModel, field_validator
+
+from aineko import AbstractNode
+from aineko.extras.connectors.secrets import inject_secrets
+
+
+class ParamsREST(BaseModel):
+    """Connector params for REST model."""
+
+    timeout: int = 10
+    url: str
+    headers: Optional[Dict[str, Any]] = None
+    data: Optional[Dict[str, Any]] = None
+    poll_interval: int = 5
+    poll_throttle: float = 0.1
+    max_retries: int = 30
+    metadata: Optional[Dict[str, Any]] = None
+    retry_sleep: float = 5
+
+    @field_validator("url")
+    @classmethod
+    def supported_url(cls, url: str) -> str:  # pylint: disable=no-self-argument
+        """Validates that the url is a valid HTTPS URL."""
+        if not (url.startswith("https://") or url.startswith("http://")):
+            raise ValueError(
+                "Invalid url provided to HTTPS params. "
+                'Expected url to start with "https://". '
+                f"Provided url was: {url}"
+            )
+        return url
+
+
+class REST(AbstractNode):
+    """Connects to an REST endpoint via HTTPS."""
```

Comment:
Can you please include an example of how to configure such a node in a yml config file?

Reply:
Yes I will document similar to WebSocketClient